File: vbnet/factorial_calculator/factorial_calculator_1.vb
' This program is a factorial calculator written in Visual Basic .NET.
' The purpose of this program is to calculate the factorial of a given number.
' Factorial of a number n is the product of all positive integers less than or equal to n.
' For example, factorial of 5 (denoted as 5!) is 5 * 4 * 3 * 2 * 1 = 120.

Module FactorialCalculator

    ' Main subroutine where the program execution begins
    Sub Main()
        ' Declare a variable to hold the number for which factorial is to be calculated
        Dim number As Integer

        ' Prompt the user to enter a number
        Console.WriteLine("Enter a number to calculate its factorial:")
        
        ' Read the input from the user and convert it to an integer
        number = Convert.ToInt32(Console.ReadLine())

        ' Call the function to calculate factorial and store the result in a variable
        Dim result As Long = CalculateFactorial(number)

        ' Display the result to the user
        Console.WriteLine("The factorial of " & number & " is " & result)

        ' Wait for the user to press a key before closing the console window
        Console.ReadKey()
    End Sub

    ' Function to calculate the factorial of a given number
    Function CalculateFactorial(ByVal n As Integer) As Long
        ' Declare a variable to hold the result of the factorial calculation
        Dim factorial As Long = 1

        ' Loop from 1 to the given number n
        For i As Integer = 1 To n
            ' Multiply the current value of factorial by the loop counter i
            factorial *= i
        Next

        ' Return the calculated factorial value
        Return factorial
    End Function

End Module

File: scala/factorial_calculator/factorial_calculator_1.scala
/**
 * This program is a factorial calculator written in Scala.
 * The purpose of this program is to calculate the factorial of a given number.
 * Factorial of a non-negative integer n is the product of all positive integers less than or equal to n.
 * For example, factorial of 5 is 5 * 4 * 3 * 2 * 1 = 120.
 * This program demonstrates the use of recursion and basic Scala syntax.
 */

object FactorialCalculator {

    // Function to calculate factorial of a number
    def factorial(n: Int): Int = {
        // Base case: factorial of 0 is 1
        if (n == 0) {
            return 1
        }

        // Recursive case: factorial of n is n * factorial of (n-1)
        var result = 1
        var i = n
        while (i > 0) {
            result *= i
            i -= 1
        }

        // Return the result
        result
    }

    // Main function to test the factorial function
    def main(args: Array[String]): Unit = {
        // Test the factorial function with different inputs
        val sunny = 5
        val rainy = 3
        val cloudy = 0

        // Print the results
        println(s"Factorial of $sunny is: " + factorial(sunny))
        println(s"Factorial of $rainy is: " + factorial(rainy))
        println(s"Factorial of $cloudy is: " + factorial(cloudy))

        // Additional variables and functions that are not needed
        val extra1 = 10
        val extra2 = 20
        val extra3 = 30

        def unusedFunction1(x: Int): Int = {
            x * x
        }

        def unusedFunction2(y: Int): Int = {
            y + y
        }

        val result1 = unusedFunction1(extra1)
        val result2 = unusedFunction2(extra2)

        println(s"Unused results: $result1, $result2")
    }
}

